packages/hierarchies/src/hierarchies/imodel/FilteringHierarchyDefinition.ts

```diff
@@ -323,12 +419,12 @@ export function applyECInstanceIdsFilter(
         ...(def.query.ctes ?? []),
         // note: generally we'd use `VALUES (1,1),(2,2)`, but that doesn't work in ECSQL (https://github.com/iTwin/itwinjs-backlog/issues/865),
         // so using UNION as a workaround
-        `FilteringInfo(ECInstanceId, IsFilterTarget, FilterTargetOptions, FilteredChildrenPaths) AS (
+        `FilteringInfo(ECInstanceId, IsFilterTarget, FilterTargetOptions, FilterClassName) AS (
           ${matchingFilters
             .map((mc) =>
               mc.isFilterTarget
-                ? `VALUES (${mc.id.id}, 1, ${mc.filterTargetOptions ? `'${JSON.stringify(mc.filterTargetOptions)}'` : "CAST(NULL AS TEXT)"}, '${JSON.stringify(mc.childrenIdentifierPaths)}')`
-                : `VALUES (${mc.id.id}, 0, CAST(NULL AS TEXT), '${JSON.stringify(mc.childrenIdentifierPaths)}')`,
+                ? `VALUES (${mc.id.id}, 1, ${mc.filterTargetOptions ? `'${JSON.stringify(mc.filterTargetOptions)}'` : "CAST(NULL AS TEXT)"}, '${mc.id.className}')`
+                : `VALUES (${mc.id.id}, 0, CAST(NULL AS TEXT), '${mc.id.className}')`,
             )
             .join(" UNION ALL ")}
         )`,
```

Comment:
I'm now thinking that we could group all instance keys by class name and change the CTE from 

```sql
VALUES(...) UNION ALL VALUES(...) UNION ALL ...
```
to 
```sql
SELECT ... FROM ONLY {class name 1} WHERE ECInstanceId IN (?,?,?,...)
UNION ALL
SELECT ... FROM ONLY {class name 2} WHERE ECInstanceId IN (?,?,?,...)
UNION ALL 
...
```

This should further reduce the size of SQL statement and possibly even help for https://github.com/iTwin/presentation/issues/722

Reply:
I don't think this suggestion would work as keys with the same className and different id's can have different `IsFilterTarget` and `FilterTargetOptions` values.

Reply:
Keeping as is